src/main/java/webserver/controllers/TodoController.java

```diff
@@ -147,13 +208,30 @@ public TodoController(TodoService todoService) {
             .build();
     };
 
+    public Callback<Request, String> deleteTodo = (request) -> {
+        Integer id = getTodoIdFromPath(request.getPath());
+        List<Integer> ids = todoService.getTodos().stream().map(todo -> todo.getId()).collect(Collectors.toList());
+
+        if (id != null && ids.contains(id) ) {
+            todoService.deleteTodo(id);
+        }
+        
+        return new Response.Builder(HTTP_HEADERS.STATUS_204)
+            .withHeader(HTTP_HEADERS.CONTENT_TYPE, "text/html; charset=utf-8")
+            .build();
+    };
+
     private String getContentLength(String body) {
         return Integer.toString(body.length());
     }
  
-    private int getTodoIdFromPath(String path) {
+    private Integer getTodoIdFromPath(String path) {
```

Comment:
Good use of using capital-i `Integer` (which can be null, in a meaningful way) over lowercase-i `int`.  I used to do this a lot too, before `Optional` came along.